backend/app/schemas/rag.py

```
@@ -0,0 +1,189 @@
+    )
+
+    @field_validator("keyword_weight")
+    @classmethod
+    def validate_weights_sum(cls, v, info):
+        """Validate that weights sum to 1.0."""
+        vector_weight = info.data.get("vector_weight", 0.7)
+        total = vector_weight + v
+        if not (0.99 <= total <= 1.01):  # Allow small floating point errors
+            raise ValueError(
+                f"vector_weight ({vector_weight}) + keyword_weight ({v}) must equal 1.0, got {total}"
+            )
```

Ruff (0.14.8)

Avoid specifying long messages outside the exception class

(TRY003)